taurus/client/json_encoder.py

```diff
@@ -140,8 +140,8 @@ def thin_dumps(obj, **kwargs):
 
     Parameters
     ----------
-    obj: BaseEntity
-        Object to dump (must be a BaseEntity because it must have uids)
+    obj:
+        Object to dump
```

Comment:
Do we have a test that verifies good behavior when we're passed a non-BaseEntity?

Reply:
I added a test for this using a dict with a string key and value.  This raises a TypeError which seems appropriate.  I'll update the PR with this check.